src/coreclr/vm/weakreferencenative.cpp

```diff
@@ -352,6 +352,147 @@ NOINLINE Object* LoadComWeakReferenceTarget(WEAKREFERENCEREF weakReference, Type
 
 //************************************************************************
 
+FCIMPL2(Object*, ComAwareWeakReferenceNative::ComWeakRefToObject, IWeakReference* pComWeakReference, INT64 wrapperId)
+{
+    FCALL_CONTRACT;
+
+    _ASSERTE(pComWeakReference != nullptr);
+
+#if defined(FEATURE_COMINTEROP) || defined(FEATURE_COMWRAPPERS)
```

Comment:
Ifdef out the whole method?

Reply:
Yes. I think once unused code in this file is removed, only eager finalizer will stay unconditionally. 
The rest will be just these two new entry points + a helper, all can be under ifdef.